### PR TITLE
feat: add json output format to CLI commands

### PR DESCRIPTION
**Pull request by alexbrdn**

### CODE REVIEW

File: internal/app/cli/serve.go [hunk 1]
```diff
@@ -93,6 +81,40 @@ func Serve(host, port string, opts ServeOptions, repo model.RepoSpec) error {
...
+		})
+
+	// collect Middlewares for the main http handler
+	var mws = getMiddlewares(opts)
+	// create a http handler
+	httpHandler := http.NewHttpHandler(handler, mws)
+	return httpHandler, nil
+}
+
+func createTmcHandler(repo model.RepoSpec, opts ServeOptions) (*http.TmcHandler, error) {
```
Comment (enricoschuetz) [inline internal/app/cli/serve.go:104]: This function is unused in the code.

File: internal/app/cli/repo.go
Comment (enricoschuetz) [inline internal/app/cli/repo.go:1]: The command `tmc repo`  (listing all repos from config file) could also support `--format json`.
Is this by purpose not to do that or just forgotten?

File: cmd/root.go [hunk 2]
```diff
@@ -101,6 +101,11 @@ func AddRepoDisambiguatorFlags(cmd *cobra.Command) {
 	_ = cmd.MarkFlagDirname("directory")
 }
 
+func AddOutputFormatFlag(cmd *cobra.Command) {
+	cmd.Flags().String("format", "plain", "output format")
```
Comment (enricoschuetz) [inline cmd/root.go:105]: The flag help text could/should also list all possible format types (plain, json).
Like it is done e.g. in `repo_add.go` ->`repoAddCmd.Flags().StringP("type", "t", "", fmt.Sprintf("type of repo to add. One of [%s]", strings.Join(repos.SupportedTypes, ", ")))
`

File: internal/app/cli/common.go [hunk 3]
```diff
@@ -28,15 +38,37 @@ func printErrs(hdr string, errs []*repos.RepoAccessError) {
 	}
 }
 
+func printJSON(js any) {
```
Comment (enricoschuetz) [inline internal/app/cli/common.go:41]: If I do e.g. a `tmc list --format json ...`  angle brackets are encoded for the repo.
```
{
    "name": "omnicorp/omnicorp/senseall",
    "author": "omnicorp",
    "manufacturer": "omnicorp",
    "mpn": "senseall",
    "repo": "\u003c.local/a2\u003e"
  },

```
I think the encoding is meant for browsers not to interpret it as HTML.
As we print it on stdout, maybe we could/should prevent this encodung for HTML.

File: internal/app/cli/list_test.go [hunk 4]
```diff
@@ -88,14 +89,44 @@ func TestList(t *testing.T) {
...
+		// when: list from the repo
+		err := List(context.Background(), repoSpec, sp, OutputFormatJSON)
+		stdout := getStdout()
+		stderr := getStderr()
+
+		// then: there is no error
+		assert.NoError(t, err)
+		// and then: stdout outputs the listable ThingModel
+		assert.Contains(t, stdout, fmt.Sprintf("\"name\": \"%s\"", listRes.Entries[0].Name))
+		assert.Contains(t, stdout, "{")
```
Comment (enricoschuetz) [inline internal/app/cli/list_test.go:129]: We should check the json output here as you did it in the other tests...
```
var actual any
err = json.Unmarshal([]byte(stdout), &actual)
assert.NoError(t, err)
```

File: internal/app/cli/list_test.go [hunk 5]
```diff
@@ -88,14 +89,44 @@ func TestList(t *testing.T) {
...
-		err := List(context.Background(), repoSpec, sp)
+		err := List(context.Background(), repoSpec, sp, OutputFormatPlain)
 		stdout := getStdout()
 		stderr := getStderr()
 
 		// then: there is no error
 		assert.NoError(t, err)
 		// and then: stdout outputs the listable ThingModel
 		assert.Contains(t, stdout, listRes.Entries[0].Name)
+		assert.NotContains(t, stdout, "{")
```
Comment (enricoschuetz) [inline internal/app/cli/list_test.go:100]: We should check that the ouptut is no json maybe by:
```
var actual any
err = json.Unmarshal([]byte(stdout), &actual)
assert.Error(t, err)
```

File: internal/app/cli/serve.go (hunk 1 above)
Comment (alexbrdn) [reply internal/app/cli/serve.go:104]: it's been a month... I don't even remember why it was there

File: cmd/root.go (hunk 2 above)
Comment (alexbrdn) [reply cmd/root.go:105]: agreed

File: internal/app/cli/common.go (hunk 3 above)
Comment (alexbrdn) [reply internal/app/cli/common.go:41]: fixed

File: internal/app/cli/list_test.go (hunk 4 above)
Comment (alexbrdn) [reply internal/app/cli/list_test.go:129]: fixed

File: internal/app/cli/list_test.go (hunk 5 above)
Comment (alexbrdn) [reply internal/app/cli/list_test.go:100]: fixed

File: internal/app/cli/repo.go
Comment (alexbrdn) [reply internal/app/cli/repo.go:1]: I believe I've done it on purpose, but I can't remember the reason now, so I added `--format json` to `repo list` as well

**Pull request merged**